rel/overlay/etc/default.ini

```diff
@@ -23,6 +23,7 @@ file_compression = snappy
 ; and/or more OS page cache hits, but they can also increase overall response
 ; time for writes when there are many attachment write requests in parallel.
 attachment_stream_buffer_size = 4096
+db_definitions = [couch_dbs, mem3_dbs, couch_replicator_dbs, cassim_dbs]
```

Comment:
> Why application environment case doesn't works? 

Because it relies on [a hack](https://github.com/apache/couchdb/blob/master/dev/boot_node.erl#L20:L21) which makes [testing harder](https://github.com/iilyak/couchdb-couch/blob/2547-fix-broken-tests/src/test_util.erl#L257:L260).

> Can we relay on that if some app wanted to define own system database it mush have specially named module with implemented behaviour?

Using convention might work but needs to be very well documented and explained.

> But still...is there no way to avoid exposing such kind of internal information in config file?

There is also a possibility to discover all modules implementing given behavior using
`lists:keyfind(behaviour, 1, <module>:module_info(attributes))`. However we would need to iterate through all modules using `code:all_loaded`. Which gives us the same problem really. Modules have to be loaded.